third_party/xla_client/pjrt_computation_client.cc

```diff
@@ -79,13 +81,26 @@ std::vector<ComputationClient::DataPtr> PjRtComputationClient::TransferToServer(
   std::vector<ComputationClient::DataPtr> datas;
   datas.reserve(tensors.size());
   for (auto& tensor : tensors) {
-    xla::Literal literal(tensor.shape);
-    tensor.populate_fn(tensor, literal.untyped_data(), literal.size_bytes());
-
     PjRtDevice* pjrt_device = StringToPjRtDevice(tensor.device);
-    std::shared_ptr<xla::PjRtBuffer> buffer =
-        client_->BufferFromHostLiteral(literal, pjrt_device).ValueOrDie();
-    buffer->GetReadyFuture().Await();
+
+    auto literal = std::make_shared<xla::Literal>(tensor.shape);
+    tensor.populate_fn(tensor, literal->untyped_data(), literal->size_bytes());
+    absl::InlinedVector<int64_t, 4> byte_strides(
```

Comment:
What does `byte_strides` hold? 
Is it the number of bytes consumed by `tensor`?

Reply:
As I understand it, stride is used in the same sense as in [PyTorch](https://pytorch.org/docs/stable/generated/torch.Tensor.stride.html):

```
Stride is the jump necessary to go from one element to the next one in the specified dimension [dim](https://pytorch.org/docs/stable/generated/torch.Tensor.dim.html#torch.Tensor.dim).
```

Not sure what the magic number `4` is here, this was copied in from PjRt. Will look into it an clarify.

Reply:
Read the doc for `InlinedVector`, not sure it was actually doing anything useful here. Replaced it with `std::vector` to remove the magic number.

Reply:
Sounds good. The reason I brought up this question is to know if we are using `byte_strides` as a storage for the "overall size of the tensor" or for determining the "number of bytes to traverse on every stride taken"? Looks like it's the latter.

Reply:
Yeah, stride is the incremental step size, not the overall size

Reply:
Ok, in that case, are you passing the "max" incremental step size in each dimension as the input parameter to this method? If so, I understand what you are doing.

Reply:
ByteStrides fills the input `strides` vector with the stride in each dimension 

https://github.com/tensorflow/tensorflow/blob/be29b5d32fd132d7f6d75c0ba2e37b7b0347444e/tensorflow/compiler/xla/shape_util.cc#L1729-L1740